core/src/scene/directionalLight.cpp

```diff
@@ -19,10 +19,16 @@ void DirectionalLight::setDirection(const glm::vec3 &_dir) {
     m_direction = glm::normalize(_dir);
 }
 
-void DirectionalLight::setupProgram( std::shared_ptr<ShaderProgram> _shader ) {
+void DirectionalLight::setupProgram(const std::shared_ptr<View>& _view, std::shared_ptr<ShaderProgram> _shader ) {
+
+    glm::vec3 direction = m_direction;
+    if (m_origin == LightOrigin::WORLD) {
+        direction = _view->getNormalMatrix() * direction;
```

Comment:
It kind of seems like it, doesn't it? The results look correct though
On May 5, 2015 9:36 AM, "Varun" <notifications@github.com> wrote:

> In core/src/scene/directionalLight.cpp
> <https://github.com/tangrams/tangram-es/pull/103#discussion_r29668168>:
>
> > @@ -19,10 +19,16 @@ void DirectionalLight::setDirection(const glm::vec3 &_dir) {
> >      m_direction = glm::normalize(_dir);
> >  }
> >
> > -void DirectionalLight::setupProgram( std::shared_ptr<ShaderProgram> _shader ) {
> > +void DirectionalLight::setupProgram(const std::shared_ptr<View>& _view, std::shared_ptr<ShaderProgram> _shader ) {
> > +
> > +    glm::vec3 direction = m_direction;
> > +    if (m_origin == LightOrigin::WORLD) {
> > +        direction = _view->getNormalMatrix() * direction;
>
> direction should be normalized here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/103/files#r29668168>.
>